src/Parallel/GlobalCache.hpp

```diff
@@ -78,6 +57,24 @@ auto make_mutable_cache_tag_storage(
                       std::vector<std::unique_ptr<Callback>>{})...);
 }
 
+template <typename ParallelComponent, typename ComponentList>
+auto get_component_if_mocked_impl() {
+  if constexpr (tmpl::list_contains_v<ComponentList, ParallelComponent>) {
+    return ParallelComponent{};
+  } else {
+    using mock_find = tmpl::find<
+        ComponentList,
+        std::is_same<get_component_being_mocked_or_default<tmpl::_1, void>,
+                     tmpl::pin<ParallelComponent>>>;
+    static_assert(
+        tmpl::size<mock_find>::value > 0,
+        "The requested parallel component (first template argument) is not a "
+        "known parallel component (second template argument) or a component "
+        "being mocked by one of those components.");
```

Comment:
Add the suggestion "Did you forget to add it to Metavariables::component_list?" I think it's very useful to give inexperienced users this hint.

Reply:
I generally prefer to avoid making suggestions unless it is very likely to be the correct answer.  I think inexperienced users are less likely to be adding their own components than either trying to fetch something that isn't actually a component or getting confused by the mocking framework.  In either of those cases if they follow that suggestion things are going to get worse.  (I've triggered this check both of those ways but never by leaving something out of component_list.)

Reply:
Ok